invisible_cities/cities/detsim_waveforms.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+
+##################################
+######### WAVEFORMS ##############
+##################################
+def bincounter(xs : np.array,
+               dx : float =1.,
+               x0 : float =0.):
+    ixs = ((xs - x0) // dx).astype(int)
+    return np.unique(ixs, return_counts=True)
+
+
+def create_waveform(times : np.array,
+                    pes   : np.array,
+                    bins  : np.array,
+                    wf_bin_time : float,
+                    nsamples    : int) -> np.array:
+    wf = np.zeros(len(bins))
+
+    if np.sum(pes)==0:
+        return wf
+    if nsamples==0:
+        sel = pes>0
+        t = np.repeat(times[sel], pes[sel])
+        t = np.clip  (t, 0, bins[-1])
+        indexes, counts = bincounter(t, wf_bin_time)
+        wf[indexes] = counts
+        return wf
+
+    sel = pes>0
+    t = np.repeat(times[sel], pes[sel])
```

Comment:
The lines:
```
    sel = pes>0
    t = np.repeat(times[sel], pes[sel])
```
could be placed above:
```
    if nsamples==0:
```
to avoid repeating them after the `if` statement.